third_party/hdf5.BUILD

```diff
@@ -45,7 +45,7 @@ cc_library(
         "src",
         "src/H5FDsubfiling",
     ],
-    linkopts = [],
+    linkopts = ["-ldl"],
```

Comment:
Let's remove this from here and add it to the target that actually needs it.

Also this is likely not cross-platform (will probably fail on macos).

Reply:
> Let's remove this from here and add it to the target that actually needs it.

Why? It makes it a bit harder on the users of the library, do you just want to save some link time?

> Also this is likely not cross-platform (will probably fail on macos).

According to https://stackoverflow.com/questions/20169660/where-is-libdl-so-on-mac-os-x there is no need to link any library to use `<dlfcn.h>`, so I guess we could just use `select`, right? Idk about windows.